### ISSUE TITLE
Add an option to always discover branches, even if they are filed as merge requests

### ISSUE THREAD

**Issue by TravDev01** [jenkinsci/gitlab-branch-source-plugin#375]
### What feature do you want to see added?

Hey, 
we have an issue with the branch discovery. Our current setting is that only branches are discovered that are not also filed as merge requests.

We have some release branches that should always be shown in the branch list.
Sometimes a merge request is created for one of the release branches to merge changes from one release to another.
With the current implementation, the branch is removed and recreated at the merge request tab, and we lose the complete build history.
Changing the branch discovery to all branches is also not a good option for us, because then we would have some duplicated jobs.

Basically, it's not so complicated to fix, but I'm unsure what the best strategy is.

1. Extend the BranchDiscoveryTrait and add an input box with branches, that always should be shown. 
2. Create a new RegexBranchDiscoveryTrait, that will only discover branches with a given regex. I'm not sure if it will work when a branch is filtered out by one trait and included by another trait.
3. Our request is too specific and we should create our own Trait in a custom plugin. But I believe that other people are also using release branches and may be facing the same issue.

Kind regards

**Comment by DanielRaapDev** [comment]
Hi. We just updated our GitLab Source Plugin to version 698 and recognized that we see all our branches now in our Jenkins Job and no more merge requests.

We have set "Discover branches Strategy" to "Only branches that are not also filed as MRs". The new field "" is empty. But regardless if we set it to `^$` or `.*` it does not change anything after a "Scan GitLab Project Now".

After restoring the previous version 694 of the plugin and a restart of Jenkins we see only branches without MRs and those are listed again.

Can you please keep the previous setting if nothing is set for "branchesAlwaysIncludedRegex"?! (I saw in #376 that nothing should happen if it is `null`)

**Comment by DanielRaapDev** [comment]
Thanks for the investigation.
